core/modules/views_ui/js/views-admin.js

```diff
@@ -298,120 +298,128 @@ Backdrop.behaviors.viewsUiSearchOptions = {
-   */
-  handleKeyup: function (event) {
-    var found, i, j, option, search, words, wordsLength, zebraClass, zebraCounter;
+    // Click on the title checks the box.
+    this.$form.on('click', 'td.title', (event) => {
+      const $target = $(event.currentTarget);
+      $target.closest('tr').find('input').trigger('click');
+    });
 
-    // Determine the user's search query. The search text has been converted to
-    // lowercase.
-    search = this.$searchBox.val().toLowerCase();
-    words = search.split(' ');
-    wordsLength = words.length;
+    const searchBoxSelector = '#edit-options-search';
+    const controlGroupSelector = 'select[name="group"]';
+    this.$form.on(
+      'formUpdated',
+      `${searchBoxSelector},${controlGroupSelector}`,
+      $.proxy(this.handleFilter, this),
+    );
 
-    // Start the counter for restriping rows.
-    zebraCounter = 0;
+    this.$searchBox = this.$form.find(searchBoxSelector);
+    this.$controlGroup = this.$form.find(controlGroupSelector);
 
-    // Search through the search texts in the form for matching text.
-    var length = this.options.length;
-    for (i = 0; i < length; i++) {
-      // Use a local variable for the option being searched, for performance.
-      option = this.options[i];
-      found = true;
-      // Each word in the search string has to match the item in order for the
-      // item to be shown.
-      for (j = 0; j < wordsLength; j++) {
-        if (option.searchText.indexOf(words[j]) === -1) {
-          found = false;
-        }
-      }
-      if (found) {
-        // Show the checkbox row, and restripe it.
-        zebraClass = (zebraCounter % 2) ? 'odd' : 'even';
-        option.$div.show();
-        option.$div.removeClass('even odd');
-        option.$div.addClass(zebraClass);
-        zebraCounter++;
-      }
-      else {
-        // The search string wasn't found; hide this item.
-        option.$div.hide();
+    /**
+     * Get a list of option labels and their corresponding divs and maintain it
+     * in memory, so we have as little overhead as possible at keyup time.
+     */
+    this.options = this.getOptions(this.$form.find('.views-filterable-option'));
+
+    // Trap the ENTER key in the search box so that it doesn't submit the form.
+    this.$searchBox.on('keypress', (event) => {
+      if (event.which === 13) {
+        event.preventDefault();
       }
-    }
-  },
+    });
+  };
 
-  /**
-   * Filter down the list of all options based on group.
-   */
-  filterGroup: function() {
-    var groupName = this.$filterGroup.val();
-    if (groupName === 'all') {
-      this.$form.find('.views-filterable-group').show();
-    }
-    else {
-      this.$form.find('.views-filterable-group').hide();
-      this.$form.find('.views-filterable-group-' + groupName).show();
-    }
-  }
-});
+  $.extend(
+    Backdrop.viewsUi.OptionsSearch.prototype,
+  /** @lends Backdrop.viewsUi.OptionsSearch# */ {
```

Comment:
What is this `@lends`? I've never come across it before.

Reply:
@klonos Not going to lie, that was completely lifted from the Drupal 10 code as noted in the first issue comment.
https://jsdoc.app/tags-lends

Reply:
Never seen this before either, but I'm generally in favor of these kinds of type hints to variables. It can really help with IDE autocomplete.